### PR TITLE
Milestone 2 friendship v1

### PR DESCRIPTION
**Pull request by Yors-git**
In this milestone I created a model Friendship to establish the associations between a user and another user for friendships and added the tests for the models and Capybara tests for the functionality.

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    expect(likes).to eq(:has_many)
+  end
+
+  it 'check correct association' do
+    expect(friendships).to eq(:has_many)
+  end
+
+  it 'check correct association' do
+    expect(inverse_friendships).to eq(:has_many)
+  end
```
Comment (SpaYco) [inline spec/models/user_spec.rb]: - [x] please make your `it` description unique for each test case

File: spec/models/friendship_spec.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    expect(inv_friendship).to_not be_valid
+  end
+
+  it 'check correct association' do
+    expect(user).to eq(:belongs_to)
+  end
+
+  it 'check correct association' do
+    expect(friend).to eq(:belongs_to)
+  end
```
Comment (SpaYco) [inline spec/models/friendship_spec.rb]: - [x] please make your `it` description unique for each test case

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

* [x] User has to accept friendship request to start friendship ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
* [x] User can reject a friendship request ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
* [x] Posts timeline includes posts written by logged-in user and their friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/)) (currently i can see posts even from people that are not in my friend list)
* [x] some of your tests aren't working 
image



### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

**Review by ebeagusamuel — APPROVED** [review]
# Project Approved :100:
#### Hey there :wave: 
You have done amazingly well in completing this milestone and implementing all the changes requested by the previous reviewer.
You can now go ahead and merge this branch and proceed with the next milestone.

#### Happy coding :+1: 
slack(@ebeagusamuel)

**Pull request merged**